### PR TITLE
fix(placement-ordering): reposition elements using push instead of swap PD-42

### PR DESCRIPTION
**Pull request by PatriciaRomaniuc**
https://illuminate.atlassian.net/browse/PD-42

### CODE REVIEW

File: packages/placement-ordering/src/ordering.js [hunk 1]
```diff
@@ -6,6 +6,17 @@ import assign from 'lodash/assign';
...
+  const remainingItems = arr.filter((item, index) => index !== fromIndex);
+
+  return [
+    ...remainingItems.slice(0, toIndex),
+    movedItem,
+    ...remainingItems.slice(toIndex)
+  ];
+}
+
 export const swap = (arr, fromIndex, toIndex) => {
```
Comment (edeustace) [inline packages/placement-ordering/src/ordering.js:20]: from the ticket - it seems that 'swap' is not to be used at all - in which case we should remove all reference to it, and adopt 'push' as the name of the action

File: packages/placement-ordering/src/__tests__/ordering.test.js [hunk 2]
```diff
@@ -49,31 +49,42 @@ describe('ordering', () => {
...
+            .toEqual(expected);
         },
-        assertSwapLabel
+        assertPushLabel
       );
 
-      assertSwap([0, 1], [2, 1, 3, 4]);
-      assertSwap([0, 1], [2, 1], [2, 3, 1, 4]);
+      assertPush([0, 1], [2, 1, 3, 4]);
+      assertPush([0, 1], [2, 1], [2, 3, 1, 4]);
```
Comment (edeustace) [inline packages/placement-ordering/src/__tests__/ordering.test.js]: can we have a look at using `it.each` for these tests? say: 

```javascript
describe('A,B,C,D', () => {
  /** 
   * read action string as 'value' to 'index' seperated by a comma
  */
  it.each`actions   | expected
              ${'D-0']} | ${ 'D,A,B,C'}
              ${'D-0,B-1']} | ${ 'D,B,A,C'}
`( "$actions => $expected" , () => {
    // parse actions, apply them and assert the result is as expected.
  });
});
```
see: https://jestjs.io/docs/en/api#testeachtablename-fn-timeout

**Review by edeustace — CHANGES_REQUESTED** [review]

File: packages/placement-ordering/src/ordering.js (hunk 1 above)
Comment (PatriciaRomaniuc) [reply packages/placement-ordering/src/ordering.js:20]: The ticket refers to the No Placement Area behavior.

If Placement Area is OFF, the current behavior with choices order: A, B, C, D:
- if user drags D over A it replaces A with D (D, B, C, A) ('swap' from ticket)

I fixed this with:
- if user drags D over A it moves D at the top (D, A, B, C) ('push' from ticket)

The swap function is used for Placement Area ON.

Comment (edeustace) [reply packages/placement-ordering/src/ordering.js:20]: ah ok - nevermind so

**Pull request merged**